controllers/keystoneapi_controller.go

```diff
@@ -699,6 +699,13 @@ func (r *KeystoneAPIReconciler) reconcileNormal(ctx context.Context, instance *k
 
 	if instance.Status.ReadyCount > 0 {
 		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	} else {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			condition.DeploymentReadyRunningMessage))
+		return ctrl.Result{RequeueAfter: time.Second * 10}, fmt.Errorf("No pod is available")
```

Comment:
I think you can just do  `return ctrl.Result{}, nil` here as this controller is watching the deployment resource, once the deploy resource changes, this will reconcile again..